Tests/Rules/AvoidMultipleTypesParameter.tests.ps1

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+BeforeAll {
+    $ruleName = "PSAvoidMultipleTypesParameter"
+
+    $settings = @{
+        IncludeRules = @($ruleName)
+    }
+}
+
+Describe 'AvoidMultipleTypesParameter' {
```

Comment:
So I would rewrite your test structure like this:

- No `Context`s
- Each test scenario is an `It`
- `Invoke-ScriptAnalyzer` is done in the `It` rather than in a `BeforeAll`
- The `It` tests the count and properties of all the violations

We could also use a few more tests:

- When no type specifiers are supplied
- When three are
- When `[ref]` is used
- When an attribute like `[ValidateSet()]` is also used